src/proof/p_visit.ml

```diff
@@ -168,13 +168,16 @@ class virtual ['hyp] map = object (self : 'self)
     let usdef dw =
       match dw.core with
       | Dvar v ->
-         begin match self#expr scx { dw with core = Opaque v } with
-         | { core = Ix n } -> [{dw with core = Dx n}]
-         | _ ->
-            Errors.warn ~at:dw "Ignored unexpandable identifier %S in BY DEF."
-                        v;
-            []
-         end
+            begin
+            let opaque = Opaque v @@ dw in
+            let opaque = self#expr scx opaque in
+            match opaque.core with
+            | Opaque _ -> [Dvar v @@ dw]
+            | _ -> Errors.bug ~at:dw
+                ("unexpected result from `self#expr` \
+                for the identifier `" ^ v ^
+                "` in a `BY DEF`")
+            end
```

Comment:
Could you explain this change? I see that the old code was moved to method `usable` in `p_anon.ml` but how is it called and why do we need this new code in addition?

Reply:
Thanks for the explanation.